File: beginner/javascript/2152.js
var input = require("fs").readFileSync("stdin", "utf8");
var lines = input.split("\n");

var n, i, h, m, o;
    
n = parseInt(lines.shift());

for(i = 0; i < n; i++){
    var valores = lines.shift().split(" ");
    h = parseInt(valores[0]);
    m = parseInt(valores[1]);
    o = parseInt(valores[2]);
    
    if(h < 10 && m < 10){
        if(o == 1)
            console.log(`0${h}:0${m} - A porta abriu!`);
        else
            console.log(`0${h}:0${m} - A porta fechou!`);
    } else if(h < 10 && m >= 10){
        if(o == 1)
            console.log(`0${h}:${m} - A porta abriu!`);
        else
            console.log(`0${h}:${m} - A porta fechou!`);
    } else if(h >= 10 && m < 10){
        if(o == 1)
            console.log(`${h}:0${m} - A porta abriu!`);
        else
            console.log(`${h}:0${m} - A porta fechou!`);
    } else if(h >= 10 && m >= 10){
        if(o == 1)
            console.log(`${h}:${m} - A porta abriu!`);
        else
            console.log(`${h}:${m} - A porta fechou!`);
    }

}

File: beginner/python/2152.py
n = int(input())

for _ in range(n):
    h, m, o = map(int, input().split())
    
    if h < 10 and m < 10:
        if o == 1:
            print("0%d:0%d - A porta abriu!" % (h, m))
        else:
            print("0%d:0%d - A porta fechou!" % (h, m))
    elif h < 10 and m >= 10:
        if o == 1:
            print("0%d:%d - A porta abriu!" % (h, m))
        else:
            print("0%d:%d - A porta fechou!" % (h, m))
    elif h >= 10 and m < 10:
        if o == 1:
            print("%d:0%d - A porta abriu!" % (h, m))
        else:
            print("%d:0%d - A porta fechou!" % (h, m))
    elif h >= 10 and m >= 10:
        if o == 1:
            print("%d:%d - A porta abriu!" % (h, m))
        else:
            print("%d:%d - A porta fechou!" % (h, m))
